net/graphsync_fetcher_test.go

```diff
@@ -522,12 +392,108 @@ type requestResponse struct {
 }
 
 type mockableGraphsync struct {
+	ctx              context.Context
+	builder          *chain.Builder
 	stubs            []requestResponse
 	receivedRequests []fakeRequest
 	store            bstore.Blockstore
 	t                *testing.T
 }
 
+func newMockableGraphsync(ctx context.Context, builder *chain.Builder, store bstore.Blockstore, t *testing.T) *mockableGraphsync {
+	return &mockableGraphsync{
+		ctx:     ctx,
+		builder: builder,
+		store:   store,
+		t:       t,
+	}
+}
+
+type mockGraphsyncLoader func(cid.Cid) (format.Node, error)
+
+func (mgs *mockableGraphsync) stubResponseWithLoader(pid peer.ID, c cid.Cid, s selector.Selector, loader mockGraphsyncLoader) {
+	var blks []format.Node
+	var responses []graphsync.ResponseProgress
+
+	linkLoader := func(lnk ipld.Link, lnkCtx ipld.LinkContext) (io.Reader, error) {
+		cid := lnk.(cidlink.Link).Cid
+		node, err := loader(cid)
+		if err != nil {
+			return nil, err
+		}
+		blks = append(blks, node)
+		return bytes.NewBuffer(node.RawData()), nil
+	}
+	root := cidlink.Link{Cid: c}
+	node, err := root.Load(mgs.ctx, ipld.LinkContext{}, ipldfree.NodeBuilder(), linkLoader)
+	if err != nil {
+		mgs.stubs = append(mgs.stubs, requestResponse{
+			fakeRequest{pid, root, s},
+			fakeResponse{errs: []error{err}},
+		})
+		return
+	}
+	visitor := func(tp ipldbridge.TraversalProgress, n ipld.Node, tr ipldbridge.TraversalReason) error {
+		responses = append(responses, graphsync.ResponseProgress{Node: n, Path: tp.Path, LastBlock: tp.LastBlock})
+		return nil
+	}
+	err = ipldbridge.TraversalProgress{
+		Cfg: &ipldbridge.TraversalConfig{
+			Ctx:        mgs.ctx,
+			LinkLoader: linkLoader,
+		},
+	}.TraverseInformatively(node, s, visitor)
+	mgs.stubs = append(mgs.stubs, requestResponse{
+		fakeRequest{pid, root, s},
+		fakeResponse{responses, []error{err}, blks},
```

Comment:
It looks like this returns a slice with a single nil error in non-error cases. Is this intended? Documentation on the `fakeResponse` could help clarify.